python/ray/data/impl/arrow_block.py

```diff
@@ -224,20 +224,18 @@ def to_pandas(self) -> "pandas.DataFrame":
         return self._table.to_pandas()
 
     def to_numpy(self, column: str = None) -> np.ndarray:
-        if not column:
-            raise ValueError(
-                "`column` must be specified when calling .to_numpy() "
-                "on Arrow blocks.")
-        if column not in self._table.column_names:
-            raise ValueError(
-                "Cannot find column {}, available columns: {}".format(
-                    column, self._table.column_names))
-        array = self._table[column]
-        if array.num_chunks > 1:
-            # TODO(ekl) combine fails since we can't concat ArrowTensorType?
-            array = array.combine_chunks()
-        assert array.num_chunks == 1, array
-        return self._table[column].chunk(0).to_numpy()
+        if column is None:
+            columns = self._table.column_names
```

Comment:
Why this semantic change? We shouldn't mix API changes with bug fixes.

Reply:
Hmm I saw the raised error as an indication of unimplemented support for what I think are the obvious semantics: `table.to_numpy()` --> 2D ndarray. And I didn't realize that these semantics were user-facing in `Dataset.to_numpy_refs()`... Was there a reason for us not supporting this initially, or are you saying that I should submit a separate PR adding support for this?

Reply:
So stacking wouldn't work in the more general case when dtypes are different across columns. I feel that would be pretty confusing, and it's better to only support conversion with an explicit column name.

Reply:
Ah that makes sense! I reverted it back to requiring a `column` argument, but I'm still sticking with the `ChunkedArray.to_numpy()` method for now until I see a failure case.